src/integrations/Http2Integration.ts

```diff
@@ -0,0 +1,275 @@
+                        [SpanTags.TRACE_LINKS]: [span.spanContext.spanId],
+                        [SpanTags.TOPOLOGY_VERTEX]: true,
+                    });
+
+                    const me = this;
+
+                    span._initialized();
+
+                    const clientRequest = request.apply(this, [headers, options]);
+
+                    const chunks: any = [];
+                    let responseHeaders: any;
+
+                    clientRequest.once('data', (chunk: any) => {
+                        if (!chunk) {
+                            return;
+                        }
+
+                        const totalSize = chunks.reduce((total: any, item: any) => item.length + total, 0);
+                        if (totalSize + chunk.length <= 10 * 1024) {
+                            chunks.push(typeof chunk === 'string' ? Buffer.from(chunk) : chunk);
+                        }
+                    });
+
+                    clientRequest.once('response', (res: any) => {
+                        responseHeaders = HTTPUtils.extractHeaders(res);
+
+                        ThundraLogger.debug(`<HTTP2Integration> On response of HTTP2 span with name ${operationName}`);
+                    });
+
+                    clientRequest.once('end', () => {
+
+                        let payload;
+                        try {
+                            payload = JSON.parse(chunks);
+                        } catch (error) {
+                            ThundraLogger.debug('<HTTP2Integration> Response is not valid JSON:', payload);
+                            payload = chunks.toString();
+                        }
+
+                        if (span) {
+                            const statusCode = responseHeaders[':status'] || 200;
+                            if (!config.disableHttp5xxError && `${statusCode}`.startsWith('5')) {
+                                span.setErrorTag(new HttpError(statusCode));
+                            }
+
+                            if (!config.disableHttp4xxError && `${statusCode}`.startsWith('4')) {
+                                span.setErrorTag(new HttpError(statusCode));
+                            }
+
+                            span.setTag(Http2Tags.HTTP2_STATUS, statusCode);
+                            if (!config.maskHttpBody && payload) {
+                                try {
+
+                                    span.setTag(Http2Tags.BODY, payload);
+                                } catch (error) {
+                                    ThundraLogger.error(
+                                        `<HTTPIntegration> Unable to get body of HTTP2 span with name ${operationName}:`,
+                                        error);
+                                }
+                            }
+
+                            ThundraLogger.debug(`<HTTP2Integration> Closing HTTP2 span with name ${operationName}`);
+                            span.closeWithCallback(me, options, [{}, payload]);
```

Comment:
I think, should not be closed by `closeWithCallback` because there is actual callback here. 
So, just `close` should be used instead